lib/vmpooler/pool_manager.rb

```diff
@@ -319,10 +319,62 @@ def _destroy_vm(vm, pool, provider)
         finish = format('%.2f', Time.now - start)
         $logger.log('s', "[-] [#{pool}] '#{vm}' destroyed in #{finish} seconds")
         $metrics.timing("destroy.#{pool}", finish)
+        get_vm_usage_labels(vm)
       end
       dereference_mutex(vm)
     end
 
+    def get_vm_usage_labels(vm)
+      return unless $config[:config]['usage_stats']
+      checkout = $redis.hget("vmpooler__vm__#{vm}", 'checkout')
+      return if checkout.nil?
+      jenkins_build_url = $redis.hget("vmpooler__vm__#{vm}", 'tag:jenkins_build_url')
+      user = $redis.hget("vmpooler__vm__#{vm}", 'token:user') || 'unauthenticated'
+      poolname = $redis.hget("vmpooler__vm__#{vm}", "template")
+
+      if jenkins_build_url
```

Comment:
I like that you're using guard clauses above to get out of situations where we shouldn't continue with the method. Another benefit of them is that they can keep the method "flatter" (I mean there's less nesting or indents).

I wouldn't hold this PR up for this, but if you're going to come in for changes anyway, this statement could be made into a guard clause like so:
```ruby
return $metrics.increment("usage.#{user}.#{poolname}") unless jenkins_build_url
```
I have a width-preference that would make me do that in two lines 🤷‍♂️ 🤷‍♀️, but you can see that this takes out one of the indents for most of the method since it's inside this if-block.

Reply:
Thanks. I've changed this to consider your suggestion and ship the simple metric if jenkins_build_url does not exist, and then explicitly return. The rest of the code gets to leave the block as a result.